src/backend/distributed/commands/statistics.c

```diff
@@ -70,6 +70,12 @@ PreprocessCreateStatisticsStmt(Node *node, const char *queryString,
 	CreateStatsStmt *stmt = castNode(CreateStatsStmt, node);
 
 	RangeVar *relation = (RangeVar *) linitial(stmt->relations);
+
+	if (!IsA(relation, RangeVar))
+	{
+		return NIL;
+	}
```

Comment:
The code comment from Postgres says that;
```c
typedef struct CreateStatsStmt
{
....
	List	   *relations;		/* rels to build stats on (list of RangeVar) */
....
} CreateStatsStmt;
```

So I think this means that the code-comment is wrong, is that correct?

----

Although above code probably won't cause crashes etc., I'd still suggest doing it that way, as what we do in the code-base in general, just to be safer.

```suggestion
	Node *relationNode = (Node *) linitial(stmt->relations);

	if (!IsA(relationNode, RangeVar))
	{
		return NIL;
	}
	
	RangeVar *relation = (RangeVar *) relationNode;
```

Reply:
> So I think this means that the code-comment is wrong, is that correct?

The code comment is a bit misleading, but they do throw an error if it's not a RangeVa

```C
case T_CreateStatsStmt:
				{
					Oid			relid;
					CreateStatsStmt *stmt = (CreateStatsStmt *) parsetree;
					RangeVar   *rel = (RangeVar *) linitial(stmt->relations);

					if (!IsA(rel, RangeVar))
						ereport(ERROR,
								(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
								 errmsg("cannot create statistics on the specified relation"),
								 errdetail("CREATE STATISTICS only supports tables, foreign tables and materialized views.")));
```